src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -13288,6 +13079,114 @@ private void LeaveQuery()
             _syntaxFactoryContext.QueryDepth--;
         }
 
+        private delegate (SyntaxToken openToken, PostSkipAction action) SkipBadTokens<TNode>(
+            LanguageParser parser, SyntaxToken openToken, SeparatedSyntaxListBuilder<TNode> builder, SyntaxKind expectedKind, SyntaxKind closeTokenKind) where TNode : GreenNode;
+
+        private SeparatedSyntaxList<TNode> ParseCommaSeparatedSyntaxList<TNode>(
+            ref SyntaxToken openToken,
+            SyntaxKind closeTokenKind,
+            Func<LanguageParser, bool> isPossibleElement,
+            Func<LanguageParser, TNode> parseElement,
+            SkipBadTokens<TNode> skipBadTokens,
+            bool allowTrailingSeparator,
+            SyntaxKind expectedKind = SyntaxKind.IdentifierToken,
+            ErrorCode? trailingSeparatorError = null) where TNode : GreenNode
```

Comment:
both the optional parameters are to handle two tiny deviations from teh standard parsing pattern.  Everything else is the same across all callers.